xrpl/wallet/main.py

```diff
@@ -15,48 +13,43 @@ class Wallet:
     details.
     """
 
-    def __init__(self: Wallet, seed: str) -> None:
+    def __init__(self: Wallet, public_key: str, private_key: str) -> None:
```

Comment:
Before this PR:


```py
from xrpl.core.keypairs import generate_seed
from xrpl.wallet import Wallet
my_wallet = Wallet(generate_seed())
```

After this PR:

```py
from xrpl.core.keypairs import generate_seed, derive_keypair
from xrpl.wallet import Wallet
my_wallet = Wallet(*derive_keypair(generate_seed()))
```

How it should be:

```py
from xrpl.wallet import Wallet
my_wallet = Wallet()
```

Reply:
To elaborate further:

It makes sense to me to have a classmethod that creates a wallet from some provided seed, and maybe another one for creating from a private key. But the default constructor should do a reasonable thing when you use it in the simplest way possible.

Reply:
With the current API, you can actually do this:
```py
from xrpl.wallet import Wallet
my_wallet = Wallet.create()
```

Reply:
Right, and that's fine, but what I'm saying is that it's backwards.

The _default constructor_ should do the most sensible thing with as few arguments as possible, and classmethods can provide more niche ways of instantiating a wallet.

Reply:
@mDuo13 is right - my change earlier here was wrong. we should be using `seed` only for now. can later add support for private key if someone needs it